apollo-backend/src/test/java/io/logz/apollo/EmergencyRollbackTest.java

```diff
@@ -0,0 +1,98 @@
+package io.logz.apollo;
+
+import io.logz.apollo.dao.DeploymentDao;
+import io.logz.apollo.dao.EnvironmentDao;
+import io.logz.apollo.helpers.RealDeploymentGenerator;
+import io.logz.apollo.helpers.StandaloneApollo;
+import io.logz.apollo.kubernetes.KubernetesMonitor;
+import io.logz.apollo.models.Deployment;
+import io.logz.apollo.models.Environment;
+import org.junit.Test;
+
+import javax.script.ScriptException;
+import java.io.IOException;
+import java.sql.SQLException;
+
+import static org.assertj.core.api.Java6Assertions.assertThat;
+
+public class EmergencyRollbackTest {
+
+    private KubernetesMonitor kubernetesMonitor;
+    private EnvironmentDao environmentDao;
+    private DeploymentDao deploymentDao;
+    private RealDeploymentGenerator realDeploymentGenerator;
+    private Deployment deployment;
+
+    public EmergencyRollbackTest() throws ScriptException, IOException, SQLException {
+        StandaloneApollo standaloneApollo = StandaloneApollo.getOrCreateServer();
+
+        kubernetesMonitor = standaloneApollo.getInstance(KubernetesMonitor.class);
+        environmentDao = standaloneApollo.getInstance(EnvironmentDao.class);
+        deploymentDao = standaloneApollo.getInstance(DeploymentDao.class);
+
+        realDeploymentGenerator = new RealDeploymentGenerator("image", "key", "value", 0);
+        deployment = realDeploymentGenerator.getDeployment();
+    }
+
+    @Test
+    public void testMonitorEmergencyRollbackOnUnlimitedConcurrencyEnvironment() {
+        Environment environment = realDeploymentGenerator.getEnvironment();
+        environmentDao.updateConcurrencyLimit(environment.getId(), -1);
+        Deployment emergencyRollback = realDeploymentGenerator.getDeployment();
+        emergencyRollback.setEmergencyRollback(true);
```

Comment:
Hmm.. this will always pass
You are generating a deployment, changing its memory representation, then pass a different deployment to the function